internal/statemachine/helper.go

```diff
@@ -756,6 +756,17 @@ func mountFromHost(targetDir, mountpoint string) (mountCmd, umountCmd *exec.Cmd)
 	return mountCmd, umountCmd
 }
 
+// mountNewFS creates a new filesystem and mounts it at the specified location
```

Comment:
this isn't creating a filesystem, its bind mounting a directory. different things!

Reply:
Agree. I'd even go as far as to rename the function to `mountTempFS` and then adjust the comment as well.